airflow/models/dagrun.py

```diff
@@ -381,11 +382,14 @@ def get_task_instance(self, task_id: str, session: Session = None) -> Optional[T
         :param session: Sqlalchemy ORM Session
         :type session: Session
         """
-        return (
+        ti = (
             session.query(TI)
             .filter(TI.dag_id == self.dag_id, TI.run_id == self.run_id, TI.task_id == task_id)
             .one_or_none()
         )
+        if ti:
+            set_committed_value(ti, 'dag_run', self)
+        return ti
```

Comment:
I think the same can be achieved with `options(joinedload(TI.dag_run))`? (See `get_task_instances()` above)

Reply:
It can - the difference here is the extra join - that would result in a more complex query, and _we know_ that the ti belongs to those dag run.